pkg/router/router_test.go

```diff
@@ -732,6 +732,81 @@ func TestConfigTemplate(t *testing.T) {
 				},
 			},
 		},
+		"valid route health check interval annotation": {
+			mustCreateWithConfig{
+				mustCreateEndpointSlice: mustCreateEndpointSlice{
+					name:        "servicer1",
+					serviceName: "servicer1",
+					addresses:   []string{"1.1.1.1", "1.1.1.2"},
+				},
+				mustCreateRoute: mustCreateRoute{
+					name:                "r1",
+					host:                "r1example.com",
+					targetServiceName:   "servicer1",
+					targetServiceWeight: 1,
+					time:                start,
+					annotations: map[string]string{
+						"router.openshift.io/haproxy.health.check.interval": "10s",
+					},
+				},
+				mustMatchConfig: mustMatchConfig{
+					section:     "backend",
+					sectionName: insecureBackendName(h.namespace, "r1"),
+					attribute:   "server",
+					value:       "inter 10s",
+				},
+			},
+		},
+		"route health check interval annotation exceeds the haproxy maximum": {
+			mustCreateWithConfig{
+				mustCreateEndpointSlice: mustCreateEndpointSlice{
+					name:        "servicer2",
+					serviceName: "servicer2",
+					addresses:   []string{"1.1.1.1", "1.1.1.2"},
+				},
+				mustCreateRoute: mustCreateRoute{
+					name:                "r2",
+					host:                "r2example.com",
+					targetServiceName:   "servicer2",
+					targetServiceWeight: 1,
+					time:                start,
+					annotations: map[string]string{
+						"router.openshift.io/haproxy.health.check.interval": "5000d",
+					},
+				},
+				mustMatchConfig: mustMatchConfig{
+					section:     "backend",
+					sectionName: insecureBackendName(h.namespace, "r2"),
+					attribute:   "server",
+					value:       "inter 2147483647ms",
+				},
+			},
+		},
+		"invalid route health check interval annotation": {
+			mustCreateWithConfig{
+				mustCreateEndpointSlice: mustCreateEndpointSlice{
+					name:        "servicer3",
+					serviceName: "servicer3",
+					addresses:   []string{"1.1.1.1", "1.1.1.2"},
+				},
+				mustCreateRoute: mustCreateRoute{
+					name:                "r3",
+					host:                "r3example.com",
+					targetServiceName:   "servicer3",
+					targetServiceWeight: 1,
+					time:                start,
+					annotations: map[string]string{
+						"router.openshift.io/haproxy.health.check.interval": "abc",
+					},
+				},
+				mustMatchConfig: mustMatchConfig{
+					section:     "backend",
+					sectionName: insecureBackendName(h.namespace, "r3"),
+					attribute:   "server",
+					value:       "inter 5000ms",
```

Comment:
This is interesting.  I would have thought it would return an empty string instead of the default because it is a SyntaxError: https://github.com/openshift/router/blob/master/pkg/router/template/template_helper.go#L345-L347

Reply:
No, it doesn't return the empty string because the template defaults it to 5000ms when the empty string is returned.  We need to consider the logged message in https://github.com/openshift/router/blob/master/pkg/router/template/template_helper.go#L346